io.openems.edge.goodwe.et/test/io/openems/edge/goodwe/et/ess/GoodWeEtBatteryInverterImplTest.java

```diff
@@ -0,0 +1,98 @@
+package io.openems.edge.goodwe.et.ess;
+
+import org.junit.Test;
+
+import io.openems.common.types.ChannelAddress;
+import io.openems.edge.bridge.modbus.test.DummyModbusBridge;
+import io.openems.edge.common.test.AbstractComponentTest.TestCase;
+import io.openems.edge.common.test.ComponentTest;
+import io.openems.edge.common.test.DummyConfigurationAdmin;
+import io.openems.edge.ess.test.DummyPower;
+import io.openems.edge.ess.test.ManagedSymmetricEssTest;
+import io.openems.edge.goodwe.et.GoodWeEtConstants;
+import io.openems.edge.goodwe.et.charger.GoodWeEtChargerPv1;
+
+public class GoodWeEtBatteryInverterImplTest {
+
+	private static final String MODBUS_ID = "modbus0";
+
+	private static final String ESS_ID = "ess0";
+	private static final ChannelAddress ESS_SET_ACTIVE_POWER_EQUALS = new ChannelAddress(ESS_ID,
+			"SetActivePowerEquals");
+	private static final ChannelAddress ESS_SOC = new ChannelAddress(ESS_ID, "Soc");
+	private static final ChannelAddress ESS_EMS_POWER_MODE = new ChannelAddress(ESS_ID, "EmsPowerMode");
+	private static final ChannelAddress ESS_EMS_POWER_SET = new ChannelAddress(ESS_ID, "EmsPowerSet");
+
+	private static final String CHARGER_ID = "charger0";
+	private static final ChannelAddress CHARGER_ACTUAL_POWER = new ChannelAddress(CHARGER_ID, "ActualPower");
+
+	@Test
+	public void test() throws Exception {
+		GoodWeEtChargerPv1 charger = new GoodWeEtChargerPv1();
+		new ComponentTest(charger) //
+				.addReference("cm", new DummyConfigurationAdmin()) //
+				.addReference("setModbus", new DummyModbusBridge(MODBUS_ID)) //
+				.activate(io.openems.edge.goodwe.et.charger.MyConfig.create() //
+						.setId(CHARGER_ID) //
+						.setEssId(ESS_ID) //
+						.setModbusId(MODBUS_ID) //
+						.setUnitId(GoodWeEtConstants.DEFAULT_UNIT_ID) //
+						.build());
+
+		GoodWeEtBatteryInverterImpl ess = new GoodWeEtBatteryInverterImpl();
+		ess.addCharger(charger);
+		new ManagedSymmetricEssTest(ess) //
+				.addReference("power", new DummyPower()) //
+				.addReference("cm", new DummyConfigurationAdmin()) //
+				.addReference("setModbus", new DummyModbusBridge(MODBUS_ID)) //
+				.addComponent(charger) //
+				.activate(MyConfig.create() //
+						.setId(ESS_ID) //
+						.setModbusId(MODBUS_ID) //
+						.setUnitId(GoodWeEtConstants.DEFAULT_UNIT_ID) //
+						.setCapacity(9_000) //
+						.setMaxBatteryPower(5_200) //
+						.setReadOnlyMode(false) //
+						.build()) //
+				.next(new TestCase("Scenario 1: (set-point is positive && set-point is lower than pv production)") //
```

Comment:
I tried to define test-cases to understand the logic you implemented based on the scenarios we defined together. Unfortunately that does not match at all. Why did the scenarios change? Maybe we can discuss this in person (in a telco) tomorrow.